api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -76,6 +76,22 @@
    */
   String TARGET_FILE_SIZE_BYTES = "target-file-size-bytes";
 
+  /**
+   * Determines if the data rewrite action should also remove non-global deletes associated with the data files.
+   * By enabling this option, any data filter specified through {@link #filter(Expression)} will be converted to
+   * an inclusive partition filter based on all the historical partition specs of the table.
+   */
+  String REMOVE_PARTITION_DELETES = "remove-partition-deletes";
```

Comment:
This just means non-global deletes, with lack of a better name.

Reply:
What about ALLOW_DELETE_REMOVAL_FILTER? Or ALLOW_SUBSET_DELETE_COMPACTION? OR [DISABLE|ENABLE]_PARTIAL_DELETES_REMOVAL?

I like the words allow / enable and disallow / disable as they seem to line up more with true meaning yes and false meaning exceptions. Seems like a double negative kind of thing or that true and false are swapped.

REMOVE_GLOBAL_DELETES is admittedly pretty clear right away (whole table), but this one is admittedly much less clear until I think about it for a while. I think it should be changed even if it no longer mirrors REMOVE_GLOBAL_DELETES or if we have to change the name of REMOVE_GLOBAL_DELETES. Maybe ALLOW_TABLE_SUBSET_DELETE_REMOVAL or REMOVE_DELETES_FROM_PARTITION_SUBSET if it has to be along partition boundaries?

If others don't find it to be unclear (without much thinking), I won't bike shed it at all. But wanted to throw some ideas out early.